node/grpc/server.go

```diff
@@ -307,6 +307,26 @@ func (s *Server) RetrieveChunks(ctx context.Context, in *pb.RetrieveChunksReques
 		s.node.Metrics.RecordRPCRequest("RetrieveChunks", "failure", time.Since(start))
 		return nil, fmt.Errorf("could not find chunks for batchHeaderHash %v, blob index: %v, quorumID: %v", batchHeaderHash, in.GetBlobIndex(), in.GetQuorumId())
 	}
+	if !s.config.EnableGnarkBundleEncoding && format == pb.ChunkEncoding_GNARK {
```

Comment:
there would be a case when gnark blob is enabled but, the client forgot to set the flag. why not make two interfaces altogether, and remove the flag?

Reply:
Having a separate interface (from `RetrieveChunks`) will still have the issue of "old-new" communication: if a new client talks to an old server (which doesn't have the new API), it will fail.